scipy/stats/_hypotests.py

```diff
@@ -663,3 +663,272 @@ def somersd(x, y=None):
+
+    Parameters
+    ----------
+    table : array_like of ints
+        A 2x2 contingency table.  Elements should be non-negative integers.
+    alternative : {'two-sided', 'less', 'greater'}, optional
+        Defines the alternative hypothesis.
+        The following options are available (default is 'two-sided'):
+
+        * 'two-sided'
+        * 'less': one-sided
+        * 'greater': one-sided
+
+    pooled : bool, optional
+        Whether to compute score statistic with pooled variance or
+        unpooled (Wald statistic)
+
+    num_it : int, optional
+        Number of iterations of the grid search. Default is 3. Must be
+        non-negative.
+
+    Returns
+    -------
+    ber : ``BarnardExactResult`` object
+            The object has attributes ``statistic`` and ``pvalue``.
+            ``statistic`` is the Z statistic with pooled (Score) or
+            unpooled (Wald) variance, depending on the user choice of
+            `pooled` param.
+            ``pvalue`` the probability of obtaining a test statistic at least
+            as extreme as the one that was actually observed, assuming that
+            the null hypothesis is true.
+
+    See Also
+    --------
+    chi2_contingency : Chi-square test of independence of variables in a
+        contingency table.
+    fisher_exact : Fisher exact test on a 2x2 contingency table.
+
+    Notes
+    -----
+    Barnard's test is an exact test used in the analysis of contingency
+    tables. It examines the association of two categorical variables and
+    is a more powerful alternative than Fisher's exact test
+    for 2x2 contingency tables.
+    When using Barnard exact test, we can assert three different null
+    hypothesis :
+
+    - :math:`H_0 : p_1 \geq p_2` versus :math:`H_1 : p_1 < p_2`,
+      with `alternative` = "less"
+
+    - :math:`H_0 : p_1 \leq p_2` versus :math:`H_1 : p_1 > p_2`,
+      with `alternative` = "greater"
+
+    - :math:`H_0 : p_1 = p_2` versus :math:`H_1 : p_1 \neq p_2`,
+      with `alternative` = "two-sided" (default one)
+
+    References
+    ----------
+    .. [1] Mehta, Cyrus & Senchaudhuri, Pralay. (2003).
+           "Conditional versus Unconditional Exact Tests for Comparing Two
+           Binomials"
+
+    Examples
+    --------
+    An example use of Barnard's test is presented in [1]_.
+
+        Consider the following example of a vaccine efficacy study
+        (Chan, 1998). In a randomized clinical trial of 30 subjects, 15 were
+        inoculated with a recombinant DNA influenza vaccine and the 15 were
+        inoculated with a placebo. Twelve of the 15 subjects in the placebo
+        group (80%) eventually became infected with influenza whereas for the
+        vaccine group, only 7 of the 15 subjects (47%) became infected. The
+        data are tabulated as a 2 x 2 table::
+
+                Vaccine  Placebo
+            Yes     7        12
+            No      8        3
+                    15        15
```

Comment:
I would add totals for the row marginals too as it shows unconditioned on rows.

```suggestion
                Vaccine  Placebo
            Yes     7        12        19
            No      8         3        11
                   15        15        30
```

Reply:
Might it be confusing for the user? I would let it how it was in the origin, without the total of the columns (15, 15 in this exemple). This add have been made unintentionally.

Reply:
I don't have strong feeling here. As you wish if no other reviewer is fine with it.

Reply:
Thank you for your review. Update in a42e0444dcd14e97a8c6ebf5ffd448ddbe1345ec